Application fails at startup if spring-data-redis is not on classpath

When starting the demo application with Hazelcast profile, the boostrap fails with a `ClassNotFoundException` complaining about missing `DataRedisAutoConfiguration` class.

```
java.lang.IllegalStateException: Failed to generate bean name for imported class 'com.vaadin.kubernetes.starter.KubernetesKitConfiguration'
        at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.registerBeanDefinitionForImportedConfigurationClass(ConfigurationClassBeanDefinitionReader.java:172) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:145) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:124) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:454) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:306) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:784) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:602) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.boot.web.server.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:143) ~[spring-boot-web-server-4.0.0-RC2.jar:4.0.0-RC2]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:765) ~[spring-boot-4.0.0-RC2.jar:4.0.0-RC2]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:454) ~[spring-boot-4.0.0-RC2.jar:4.0.0-RC2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:321) ~[spring-boot-4.0.0-RC2.jar:4.0.0-RC2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1374) ~[spring-boot-4.0.0-RC2.jar:4.0.0-RC2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-4.0.0-RC2.jar:4.0.0-RC2]
        at com.vaadin.kubernetes.demo.Application.main(Application.java:27) ~[classes/:na]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
        at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:52) ~[spring-boot-devtools-4.0.0-RC2.jar:4.0.0-RC2]
Caused by: java.lang.IllegalArgumentException: Could not find class [org.springframework.boot.data.redis.autoconfigure.DataRedisAutoConfiguration]
        at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:353) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.core.annotation.TypeMappedAnnotation.adapt(TypeMappedAnnotation.java:451) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.core.annotation.TypeMappedAnnotation.getValue(TypeMappedAnnotation.java:384) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.core.annotation.TypeMappedAnnotation.asMap(TypeMappedAnnotation.java:273) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.core.annotation.AbstractMergedAnnotation.asAnnotationAttributes(AbstractMergedAnnotation.java:191) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.AnnotationBeanNameGenerator.determineBeanNameFromAnnotation(AnnotationBeanNameGenerator.java:143) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.AnnotationBeanNameGenerator.generateBeanName(AnnotationBeanNameGenerator.java:110) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.registerBeanDefinitionForImportedConfigurationClass(ConfigurationClassBeanDefinitionReader.java:168) ~[spring-context-7.0.0-RC3.jar:7.0.0-RC3]
        ... 18 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.boot.data.redis.autoconfigure.DataRedisAutoConfiguration
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na]
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[na:na]
        at java.base/java.lang.Class.forName0(Native Method) ~[na:na]
        at java.base/java.lang.Class.forName(Class.java:534) ~[na:na]
        at java.base/java.lang.Class.forName(Class.java:513) ~[na:na]
        at org.springframework.boot.devtools.restart.classloader.RestartClassLoader.loadClass(RestartClassLoader.java:123) ~[spring-boot-devtools-4.0.0-RC2.jar:4.0.0-RC2]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[na:na]
        at java.base/java.lang.Class.forName0(Native Method) ~[na:na]
        at java.base/java.lang.Class.forName(Class.java:534) ~[na:na]
        at java.base/java.lang.Class.forName(Class.java:513) ~[na:na]
        at org.springframework.util.ClassUtils.forName(ClassUtils.java:302) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:343) ~[spring-core-7.0.0-RC3.jar:7.0.0-RC3]
        ... 25 common frames omitted
```


The issue seems to be that the Kit autoconfiguration class has a reference to that class in the `@AutoConfigureAfter` annotation.
The fix should be to reference the class by string name.

--- a/kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/KubernetesKitConfiguration.java
+++ b/kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/KubernetesKitConfiguration.java
@@ -65,7 +65,7 @@ import com.vaadin.kubernetes.starter.sessiontracker.serialization.debug.Serializ
  */
 @AutoConfiguration
 @ConditionalOnProperty(name = "auto-configure", prefix = KubernetesKitProperties.PREFIX, matchIfMissing = true)
-@AutoConfigureAfter(value = { SpringBootAutoConfiguration.class })
+@AutoConfigureAfter(SpringBootAutoConfiguration.class)
 @EnableConfigurationProperties({ KubernetesKitProperties.class,
         SerializationProperties.class })
 public class KubernetesKitConfiguration {
@@ -269,8 +269,7 @@ public class KubernetesKitConfiguration {
     }
 
     @AutoConfiguration
-    @ConditionalOnClass({ RedisConnectionFactory.class,
-            DataRedisAutoConfiguration.class })
+    @ConditionalOnClass(DataRedisAutoConfiguration.class)
     @AutoConfigureAfter(DataRedisAutoConfiguration.class)
     public static class RedisConfiguration {
         @Bean

